### DISCUSSION TITLE
to use Cypress.require within cy.origin error

### DISCUSSION THREAD

**Discussion by SemaEfe** [cypress-io/cypress#28020] [Questions and Help]
I need use cy.origin and also I want to use my data from my page that I store web elements locator in it. However I can not use them inside the cy.origin. I can not call date from page inside cy.origin. And I searched a solition then I found use cy.require it can be useful. I tried below scripts but still not working. It can be wrong useage, please help me. Thanks

this is my page that I store web element locators and I want to call them inside cy.origin;

     class TradeinPage{

    getTradein_button(){
        return cy.get('.primary--text')
    }
    getEmailoButtonPDF(){
        return cy.get('.v-dialog > :nth-child(1) > .d-flex > :nth-child(2)')
      }
}
    export default TradeinPage

And this is a my test page:

     cy.visit('https://cash.com')

     cy.get('#username').type('Carla').click()

     cy.origin('https://cash/statistics/',()=> { 

     const trade = Cypress.require('../../../pages/TradeinPage') //This path is where I store my locators and I want to call them

      cy.get(trade.getTradein_button).click() // in here it is not see page object

**Comment by portyu9** [comment]
Hi @SemaEfe 

There are a couple of separate issues in that example.

First, `cy.origin()` should identify the **origin**, not an application path. For example, if the secondary origin really is `https://cash`, use:

```js
cy.origin('https://cash', ...)
```

rather than:

```js
cy.origin('https://cash/statistics/', ...)
```

Second, this is incorrect:

```js
cy.get(trade.getTradein_button).click()
```

`getTradein_button` is a method that already calls `cy.get()`. It is not a selector string.

For this particular page object, the simplest solution is to pass the selector as serializable data:

```js
const tradeinSelector = '.primary--text'

cy.origin(
  'https://cash',
  { args: { tradeinSelector } },
  ({ tradeinSelector }) => {
    cy.get(tradeinSelector).click()
  }
)
```

That avoids the dependency-loading problem entirely.

If you really do need the page-object class inside the secondary origin, current Cypress supports that with `Cypress.require()`, but you must enable:

```js
e2e: {
  experimentalOriginDependencies: true,
}
```

Then call `Cypress.require()` **inside** the `cy.origin()` callback using a static string path, and instantiate/use the exported class there. Cypress's current `cy.origin()` documentation specifically shows this pattern for page objects.

For a file that only stores selectors, however, passing plain serializable selectors through `args` is simpler and avoids the experimental dependency feature.

Please mark this answer as accepted if it helped, thank you.
